### PR TITLE
Divide IconScoreContext class

### PR DESCRIPTION
**Pull request by leeheonseung**
## Overview
currently IconScoreContext class is constituted data_struct and link system SCORE API (#8)

so someone who develop SCORE can accept API function which can't use
therefore I try to refactoring code about IconScoreContext Class

TODO. Divide that class to data class and API class

### Describe
IconScoreContext : Contains the useful information to process user's jsonrpc request like invoke and query, and have tools which can access governance api

prev)
IconScoreContext <- data struct + API

after)
IconContext <- data struct
IconContextUtil <- API

IconContextUtil only have static method for using context API

### CODE REVIEW

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 1]
```diff
@@ -19,10 +19,11 @@
 from typing import TYPE_CHECKING, Callable
 
 from iconcommons import Logger
+from iconservice.iconscore.icon_score_context_util import IconScoreContextUtil
```
Comment (sink772) [inline iconservice/deploy/icon_score_deploy_engine.py]: Need to unify the import convention like the following from all modified files.
```suggestion
from .icon_score_context_util import IconScoreContextUtil
```

File: iconservice/iconscore/icon_score_engine.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 from typing import TYPE_CHECKING
 
+from iconservice.iconscore.icon_score_context_util import IconScoreContextUtil
```
Comment (leeheonseung) [inline iconservice/iconscore/icon_score_engine.py]: check

File: iconservice/iconscore/icon_system_score_base.py [hunk 3]
```diff
@@ -16,9 +16,9 @@
 
 from abc import abstractmethod
 
-from typing import TYPE_CHECKING, Optional
-
+from typing import TYPE_CHECKING, Optional, Type
 
+from iconservice.iconscore.icon_score_context_util import IconScoreContextUtil
```
Comment (leeheonseung) [inline iconservice/iconscore/icon_system_score_base.py]: check

File: iconservice/deploy/icon_score_deploy_storage.py [hunk 4]
```diff
@@ -18,6 +18,7 @@
 from struct import pack, unpack
 from typing import TYPE_CHECKING, Optional, Tuple
 
+from iconservice.iconscore.icon_score_context_util import IconScoreContextUtil
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deploy_storage.py]: check

File: iconservice/icon_service_engine.py [hunk 5]
```diff
@@ -20,6 +20,7 @@
 from typing import TYPE_CHECKING, List, Any, Optional
 
 from iconcommons.logger import Logger
+from iconservice.iconscore.icon_score_context_util import IconScoreContextUtil
```
Comment (leeheonseung) [inline iconservice/icon_service_engine.py]: check

File: iconservice/iconscore/icon_score_base.py [hunk 6]
```diff
@@ -21,6 +21,7 @@
 from functools import partial, wraps
 from typing import TYPE_CHECKING, Callable, Any, List, Tuple, Optional, Union
 
+from iconservice.iconscore.icon_score_context_util import IconScoreContextUtil
```
Comment (leeheonseung) [inline iconservice/iconscore/icon_score_base.py]: check

File: iconservice/iconscore/icon_system_score_base.py [hunk 7]
```diff
@@ -53,11 +53,19 @@ def __init__(self, db: 'IconScoreDatabase') -> None:
     def is_builtin_score(self, score_address: 'Address') -> bool:
         return util_is_builtin_score(str(score_address))
 
+    # TODO remove after Update 0.0.6
+    def get_icon_service_flag(self) -> int:
+        return IconScoreContextUtil.icon_service_flag
+
+    def get_icon_score_context_util(self) -> Type['IconScoreContextUtil']:
```
Comment (goldworm-icon) [inline iconservice/iconscore/icon_system_score_base.py]: I think that IconScoreContextUtil should be used in IconScoreBase or IconSystemScoreBase classes only.
IconScoreContextUtil should not be exposed to SCORE subclasses directly.

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 8]
```diff
@@ -191,7 +192,7 @@ def _on_deploy_for_builtin(self,
         """Install an icon score for builtin
         """
 
-        score_root_path = context.icon_score_mapper.score_root_path
+        score_root_path = IconScoreContextUtil.icon_score_mapper.score_root_path
```
Comment (goldworm-icon) [inline iconservice/deploy/icon_score_deploy_engine.py]:
```suggestion
        score_root_path: str = IconScoreContextUtil.get_score_root_path(context)
```

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 9]
```diff
@@ -145,7 +146,7 @@ def _score_deploy(self, context: 'IconScoreContext', tx_params: 'IconScoreDeploy
         content_type = data.get('contentType')
 
         if content_type == 'application/tbears':
-            if not context.legacy_tbears_mode:
+            if not IconScoreContextUtil.legacy_tbears_mode:
```
Comment (goldworm-icon) [inline iconservice/deploy/icon_score_deploy_engine.py]: I can not find the reason why "legacy_tbears_mode" should be moved from IconScoreContext to IconScoreContextUtil.

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 10]
```diff
@@ -213,7 +214,7 @@ def _on_deploy_for_builtin(self,
             pass
 
         try:
-            score = context.icon_score_mapper.load_score(score_address, next_tx_hash)
+            score = IconScoreContextUtil.icon_score_mapper.load_score(score_address, next_tx_hash)
```
Comment (goldworm-icon) [inline iconservice/deploy/icon_score_deploy_engine.py]:
```suggestion
            score = IconScoreContextUtil.load_score(context, score_address, next_tx_hash)
```
Preserving icon_score_mapper in IconScoreContext

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 11]
```diff
@@ -223,7 +224,7 @@ def _on_deploy_for_builtin(self,
             Logger.warning('revert to add wait icon score', ICON_DEPLOY_LOG_TAG)
             raise e
 
-        context.icon_score_mapper.put_score_info(score_address, score, next_tx_hash)
+        IconScoreContextUtil.icon_score_mapper.put_score_info(score_address, score, next_tx_hash)
```
Comment (goldworm-icon) [inline iconservice/deploy/icon_score_deploy_engine.py]:
```suggestion
        IconScoreContextUtil.put_score_info(context, score_address, score, next_tx_hash)
```

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 12]
```diff
@@ -252,7 +253,7 @@ def _on_deploy(self,
             next_tx_hash = bytes(DEFAULT_BYTE_SIZE)
 
         if content_type == 'application/tbears':
-            score_root_path = context.icon_score_mapper.score_root_path
+            score_root_path = IconScoreContextUtil.icon_score_mapper.score_root_path
```
Comment (goldworm-icon) [inline iconservice/deploy/icon_score_deploy_engine.py]:
```suggestion
            score_root_path: str = IconScoreContextUtil.get_score_root_path(context)
```

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 13]
```diff
@@ -304,7 +305,7 @@ def _on_deploy(self,
             context.msg = backup_msg
             context.tx = backup_tx
 
-        context.icon_score_mapper.put_score_info(score_address, score, next_tx_hash)
+        IconScoreContextUtil.icon_score_mapper.put_score_info(score_address, score, next_tx_hash)
```
Comment (goldworm-icon) [inline iconservice/deploy/icon_score_deploy_engine.py]:
```suggestion
        IconScoreContextUtil.put_score_info(context, score_address, score, next_tx_hash)
```

File: iconservice/iconscore/icon_system_score_base.py [hunk 14]
```diff
@@ -53,11 +53,19 @@ def __init__(self, db: 'IconScoreDatabase') -> None:
...
+        return IconScoreContextUtil
+
     def get_deploy_tx_params(self, tx_hash: bytes) -> Optional['IconScoreDeployTXParams']:
-        return self._context.icon_score_deploy_engine.icon_deploy_storage.get_deploy_tx_params(self._context, tx_hash)
+        return IconScoreContextUtil.icon_score_deploy_engine.icon_deploy_storage.get_deploy_tx_params(self._context,
```
Comment (goldworm-icon) [inline iconservice/iconscore/icon_system_score_base.py]:
```suggestion
        return IconScoreContextUtil.get_deploy_tx_params(self._context, tx_hash)
```

File: iconservice/iconscore/icon_system_score_base.py [hunk 15]
```diff
@@ -53,11 +53,19 @@ def __init__(self, db: 'IconScoreDatabase') -> None:
...
+                                                                                                      tx_hash)
 
     def get_deploy_info(self, address: 'Address') -> Optional['IconScoreDeployInfo']:
-        return self._context.icon_score_deploy_engine.icon_deploy_storage.get_deploy_info(self._context, address)
+        return IconScoreContextUtil.icon_score_deploy_engine.icon_deploy_storage.get_deploy_info(self._context,
```
Comment (goldworm-icon) [inline iconservice/iconscore/icon_system_score_base.py]:
```suggestion
        return IconScoreContextUtil.get_deploy_info(self._context, address)
```

File: iconservice/iconscore/icon_system_score_base.py (hunk 7 above)
Comment (leeheonseung) [reply iconservice/iconscore/icon_system_score_base.py]: I understand
first of all, removed get_icon_score_context_util function and made specific functions to replace logics which use get_icon_score_context_util

File: iconservice/deploy/icon_score_deploy_engine.py (hunk 9 above)
Comment (leeheonseung) [reply iconservice/deploy/icon_score_deploy_engine.py]: I understand.
rollback this

**Pull request merged**